File: diamondinheri.java
class A {
    void display() {
        System.out.println("A");
    }
}

class B extends A {
   
    void display() {
        System.out.println("B");
    }
}

class C extends A {
    void display() {
        System.out.println("C");
    }
}

class D extends B {
}

public class Diamond{
    public static void main(String[] args) {
        D d = new D();
        d.display(); 
    }
}

File: diamondinheri.cpp
#include<iostream>
using namespace std;
class A {
    public :
    void display() {
        cout<<"A"<<endl;
    }
};

class B : public A {
   public:
    void display() {
        cout<<"B"<<endl;
    }
};

class C:  public  A {
  public:  void display() {
        cout<<"C"<<endl;
    }
};

class D : public B,public C {
};

int main(){
        D obj;
        obj.B::display();
        obj.C::display(); 
}

